pkg/version/version.go

```diff
@@ -0,0 +1,307 @@
+
+// CreateFirstVersion works much likst CreateVersion except that it assumes version 0
+// and never attempts to calculate a diff, or look at previous versions
+func CreateFirstVersion(appID string, filesInDir string, source string) (int64, error) {
+	return createVersion(appID, filesInDir, source, nil)
+}
+
+// CreateVersion creates a new version of the app in the database, but the caller
+// is responsible for uploading the archive to s3
+func CreateVersion(appID string, filesInDir string, source string, currentSequence int64) (int64, error) {
+	return createVersion(appID, filesInDir, source, &currentSequence)
+}
+
+// this is the common, internal function to create an app version, used in both
+// new and updates to apps
+func createVersion(appID string, filesInDir string, source string, currentSequence *int64) (int64, error) {
+	kotsKinds, err := kotsutil.LoadKotsKindsFromPath(filesInDir)
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to read kots kinds")
+	}
+
+	supportBundleSpec, err := kotsKinds.Marshal("troubleshoot.replicated.com", "v1beta1", "Collector")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal support bundle spec")
+	}
+	analyzersSpec, err := kotsKinds.Marshal("troubleshoot.replicated.com", "v1beta1", "Analyzer")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal analyzer spec")
+	}
+	preflightSpec, err := kotsKinds.Marshal("troubleshoot.replicated.com", "v1beta1", "Preflight")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal preflight spec")
+	}
+
+	appSpec, err := kotsKinds.Marshal("app.k8s.io", "v1beta1", "Application")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal app spec")
+	}
+	kotsAppSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Application")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal kots app spec")
+	}
+	backupSpec, err := kotsKinds.Marshal("velero.io", "v1", "Backup")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal backup spec")
+	}
+
+	licenseSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "License")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal license spec")
+	}
+	configSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Config")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal config spec")
+	}
+	configValuesSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "ConfigValues")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal configvalues spec")
+	}
+
+	// TODO: solve
+
+	registrySettings, err := getRegistrySettingsForApp(appID)
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to get registry settings")
+	}
+
+	preflightBytes, err := render.RenderFile(kotsKinds, registrySettings, []byte(preflightSpec))
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to template preflight spec")
+	}
+	templatedPreflight := string(preflightBytes)
```

Comment:
is this still required, with the new inline preflight caller?

Reply:
hmmm, good question..

Reply:
actually why do we even store them in the database?  i'm going to start storing the unrendered version, but not change the schema to drop that column for now.

Reply:
because the TS read the preflight specs from the database